Snippets/PostgreSqlTransport/PostgreSqlTransport_8/ConnectionStrings.cs

```diff
@@ -12,6 +12,40 @@ void ConnectionString(EndpointConfiguration endpointConfiguration)
         #endregion
     }
 
+    void ConnectionEntra(EndpointConfiguration endpointConfiguration)
+    {
+        #region postgresql-config-entra
+
+        var connection = "Server=test.postgres.database.azure.com;Database=postgres;Port=5432;User Id=<entra user id>;Ssl Mode=Require;";
+        var dataSourceBuilder = new NpgsqlDataSourceBuilder(connection);
+        if (string.IsNullOrEmpty(dataSourceBuilder.ConnectionStringBuilder.Password))
+        {
+            dataSourceBuilder.UsePeriodicPasswordProvider(async (_, ct) =>
+            {
+                var credentials = new DefaultAzureCredential();
+                var token = await credentials.GetTokenAsync(new TokenRequestContext(["https://ossrdbms-aad.database.windows.net/.default"]), ct);
+                return token.Token;
+            }, TimeSpan.FromHours(24), TimeSpan.FromSeconds(10));
+        }
+        var builder = dataSourceBuilder.Build();
+        var transport = new PostgreSqlTransport(async cancellationToken =>
+        {
+            var connection = builder.CreateConnection();
+            try 
+            {
+                await connection.OpenAsync(cancellationToken);
```

Comment:
@danielmarbach I assume the open is required when utilising the connection factory, since all the samples show it, however it doesn't appear that the documentation doesn't specify that it's required. Is it actually required? Am I missing the documentation that states it is?

Reply:
The connection factory is responsible to open a new connection. The code assumes the connection is open so the snippet is correct. See for example https://github.com/Particular/NServiceBus.SqlServer/blob/ba8c6a15b30eb80ab1ca2c52d8cf90c362f072aa/src/NServiceBus.Transport.Sql.Shared/PubSub/SubscriptionTable.cs#L34

but yeah the explanation doesn't really say that https://docs.particular.net/transports/sql/connection-settings#custom-sql-server-transport-connection-factory